### PR TITLE
chore(query-builder): Remove feature flag checks for issue search bars

### PR DESCRIPTION
**Pull request by malwilley**
The new search is GA, this removes the feature flag check for the issue search bars and deletes a lot of now-unnecessary code.

### CODE REVIEW

File: static/app/stores/tagStore.tsx
Comment (malwilley) [inline static/app/stores/tagStore.tsx:1]: The issue-specific functions are no longer used here, that is all done in `useFetchIssueTags` now. This store is still being used by some other search bars though so can't fully delete

File: static/app/utils/withIssueTags.tsx
Comment (malwilley) [inline static/app/utils/withIssueTags.tsx:1]: overview.tsx doesn't need to deal with tags anymore, that logic is in the search bar itself now

File: static/app/views/issueDetails/groupEvents.tsx
Comment (malwilley) [inline static/app/views/issueDetails/groupEvents.tsx:1]: This file needed a lot of changes to keep tests passing, mostly to do with adding awaits and using different actions to interact with the search bar

File: static/app/views/issueList/overview.spec.tsx
Comment (malwilley) [inline static/app/views/issueList/overview.spec.tsx:1]: This file needed a lot of changes to keep tests passing, mostly to do with adding awaits and using different actions to interact with the search bar

File: static/app/views/issueList/searchBar.spec.tsx [hunk 1]
```diff
@@ -35,145 +26,9 @@ describe('IssueListSearchBar', function () {
     MockApiClient.clearMockResponses();
   });
 
-  describe('updateAutoCompleteItems()', function () {
```
Comment (malwilley) [inline static/app/views/issueList/searchBar.spec.tsx:38]: Old tests, already have ones that test the new behavior

**Review by scttcper — APPROVED** [review]
:feelsgood:

**Pull request merged**